package/MDAnalysis/core/selection.py

```diff
@@ -1381,3 +1356,55 @@ def _parse_subexp(self):
 
 # The module level instance
 Parser = SelectionParser()
+
+# create a module container to avoid name clashes of autogenerated classes
+selection_classes = types.ModuleType(f'{__name__}.selection_classes')
```

Comment:
I don't like the name "selection_classes" as it talks about implementation details – MDAnalysis.core.selection.selection_classes – maybe make it underscored (to indicate that users should not bother with it as it's internal and can change) and call it, I don't know, *core.selection._components* or *core.selection._selectors*....

Reply:
Will this module be auto-documented or does it even make sense to autodoc it? It would provide an overview over which selections are available.

Reply:
Or `core.selection._selections`?

Reply:
Normal selection classes aren't documented (instead, they're manually added to `selections.rst`) and I don't think most users would need them to be. At least in the user guide, "selectable attributes + generic example" could be another scripted section? But it is likely one of those things that benefits a lot from flexibly manual documentation. "resname CYS TYR" and "altLoc A" is much nicer than "resname GenericString" and "altLoc GenericString".